cmd/private-org-sync/main.go

```diff
@@ -398,6 +392,67 @@ func (g gitSyncer) mirror(repoDir string, src, dst location) error {
 	return nil
 }
 
+func addGitRemote(logger *logrus.Entry, git gitFunc, org, repo, repoDir, remoteName string) error {
+	remoteSetupLogger := logger.WithField("remote-name", remoteName)
+	remoteSetupLogger.Debug("Remote does not exist, setting up")
+
+	srcUrl, err := url.Parse(fmt.Sprintf("https://github.com/%s/%s", org, repo))
+	if err != nil {
+		remoteSetupLogger.WithError(err).Error("Failed to construct URL for the source remote")
+		return fmt.Errorf("failed to construct URL for the source remote")
+	}
+
+	remoteSetupLogger = remoteSetupLogger.WithField("remote-url", srcUrl.String())
+	remoteSetupLogger.Debug("Adding remote")
+
+	if _, exitCode, err := git(logger, repoDir, "remote", "add", remoteName, srcUrl.String()); err != nil || exitCode != 0 {
+		remoteSetupLogger.WithField("exit-code", exitCode).WithError(err).Error("Failed to set up source remote")
+		return fmt.Errorf("failed to set up source remote")
+	}
+
+	return nil
+}
+
+func mergeRemotesAndPush(logger *logrus.Entry, git gitFunc, repoDir, srcRemote, destRemote, org, repo, branch, destUrl string, confirm bool) error {
+	if err := addGitRemote(logger, git, org, repo, repoDir, destRemote); err != nil {
+		return err
+	}
+
+	sourceBranch := fmt.Sprintf("%s/%s", srcRemote, branch)
+	if _, _, err := git(logger, repoDir, []string{"merge", sourceBranch}...); err != nil {
+		return fmt.Errorf("failed to merge %s: %v", sourceBranch, err)
+	}
+
+	// Create another branch, fetch destination and merge
+	privBranch := fmt.Sprintf("%s-priv", branch)
+	if _, _, err := git(logger, repoDir, []string{"checkout", "-b", privBranch}...); err != nil {
+		return fmt.Errorf("failed to checkout to %s: %v", privBranch, err)
+	}
+
+	if _, _, err := git(logger, repoDir, []string{"fetch", destRemote}...); err != nil {
```

Comment:
`fetch remote` fetches all branches but we only need one, sot this should be `fetch remote branch`.